### PR TITLE
TASK SIO-4213 Support custom plans

### PR DESCRIPTION
**Pull request by rincler**

### CODE REVIEW

**Review by vss414 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -67,6 +67,20 @@ The module gives the ability to select a specific operating system on ordering o
...
+#### Plan params
+
+The module gives the ability to select specific plan params such as `VCPU`, `Memory`, `Disk`, `Total traffic limit monthly` (for KVM and VZ), `VCPU Units`, `VCPU Limit`, `IO Priority`, `Swap` (for VZ).
+This is done in the form of configurable options.
+
+![VCPU](./docs/option-vcpu.png)
+
+Create a configurable option named one of: `VCPU`, `Memory`, `Disk`, `Total traffic limit monthly`, `VCPU Units`, `VCPU Limit`, `IO Priority`, `Swap`. Add options with the following convention: `planParamValue|planParamTitle`. This will take priority over plan params of selected plan in product module settings.
+
+`planParamValue` for `Memory` and `Swap` set in MiB.
```
Comment (oneumyvakin) [inline README.md:79]: Please provide an example.

**Review by cyxrop — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -67,6 +67,20 @@ The module gives the ability to select a specific operating system on ordering o
...
+`planParamValue` for `Disk` set in GiB.
+`planParamValue` for `Total traffic limit monthly` set in units which are specified in total traffic limit option of selected plan in product module settings.
+`Total traffic limit monthly` option can only be used if plan has total traffic limit type and reducing network bandwidth is enabled.
```
Comment (avolovplesk) [inline README.md:82]: `Total traffic limit monthly` option can only be used if the plan selected in Module Settings for a product has Total Traffic limit enabled in SolusIO.

**Review by avolovplesk — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -67,6 +67,24 @@ The module gives the ability to select a specific operating system on ordering o
...
+
+![VCPU](./docs/option-vcpu.png)
+
+Create a configurable option named one of: `VCPU`, `Memory`, `Disk`, `Total traffic limit monthly`, `VCPU Units`, `VCPU Limit`, `IO Priority`, `Swap`. Add options with the following convention: `planParamValue|planParamTitle`. This will take priority over plan params of selected plan in product module settings.
+
+`planParamValue` for `Memory` and `Swap` set in MiB (e.g. `4096|4 GiB`).
+
+`planParamValue` for `Disk` set in GiB (e.g. `20|20 GiB`).
+
+`planParamValue` for `Total traffic limit monthly` set in units which are specified in Total Traffic limit option of the plan selected in Module Settings for a product in SolusIO (e.g. `1|1 GiB` if Total Traffic limit unit is `GiB` or `1024|1 Gib` if Total Traffic limit unit is `MiB`).
```
Comment (GolovizninVV) [inline README.md:83]: `planParamValue` for the `Total traffic limit monthly` set in units, which are specified in the `Total Traffic limit` option of the plan selected in the Module Settings of a product in SolusIO (e.g. `1|1 GiB` if the `Total Traffic limit` unit is `GiB` or `1024|1 Gib` if the `Total Traffic limit` unit is `MiB`).

File: README.md [hunk 4]
```diff
@@ -67,6 +67,24 @@ The module gives the ability to select a specific operating system on ordering o
...
+`Total traffic limit monthly` option can only be used if the plan selected in Module Settings for a product has Total Traffic limit enabled in SolusIO.
```
Comment (GolovizninVV) [inline README.md:84]: `Total traffic limit monthly` option can only be used if the plan selected in the Module Settings of a product has the `Total Traffic limit` option enabled in SolusIO.

**Review by GolovizninVV — CHANGES_REQUESTED** [review]

**Review by GolovizninVV — APPROVED** [review]

**Pull request merged**